sound/soc/soc-pcm.c

```diff
@@ -2023,7 +2023,7 @@ int dpcm_be_dai_hw_params(struct snd_soc_pcm_runtime *fe, int stream)
 		dev_dbg(be->dev, "ASoC: hw_params BE %s\n",
 			be->dai_link->name);
 
-		ret = __soc_pcm_hw_params(be, be_substream, &hw_params);
+		ret = __soc_pcm_hw_params(be, be_substream, &be->dpcm[stream].hw_params);
```

Comment:
@plbossart, imho no one should save the hw_params pointer passed in hw_params() callback and afaik no one does that (with the exception of sdw intel between 5.10 and the previous patch).
From other, close to ALSA core operations it is clear that the params parameter should be only used within the hw_param() callback, it is only valid during that time.

So, I'm not sure if we want to have mixed validity of the params in driver's hw_params depending where they are in the stack.

Reply:
I hear you, the point is not to encourage people to play with the argument. 

I thought using the heap also makes it easier to debug memory corruption issues. The stack is just awful to deal with.

Reply:
anyways I removed this patch